### ISSUE TITLE
Catch internal vars

### ISSUE THREAD

**Issue by JProphet89** [realm/SwiftLint#3607]
### New Issue Checklist

- [x ] Updated SwiftLint to the latest version
- [x ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The ideia is to find internal vars on the code, but for some reason is not catching group 4, does SwiftLint support name grouping like ?<internal> and on yml capture_group: "internal"?

##### Complete output when running SwiftLint, including the stack trace and command used

original regex
```regex
(.*\n)*?class\s\w*\s?[:](\s+\w*[:, {])*\{\n(.*\n)*?((\s{4}|\t)(internal\s)?((weak|lazy)\s)?(var|let))
```

```swift
//
//  Copyright © whatever
//

import Foundation

open class SomeViewModel: NSObject {

    open var someService: SomeService

    public init(someService: SomeService) {
        self.someService = SomeService
    }

    //internal var stuff = 1 // <--if uncommented should catch this one 

    open var stuff12 = 1

    internal var stuff = 1 // <--should catch this one 
}
```

### Environment

* SwiftLint running on Xcode with custom rules 

```yml
internal_acl: 
      included: ".*((View|ViewController|ViewModel|Cell).swift)\\b$"
      name: "Delivery Internal access level"
      regex: "(.*\\n)*?class\\s\\w*\\s?[:](\\s+\\w*[:, {])*\\{\\n(.*\\n)*?((\\s{4}|\\t)(internal\\s)?((weak|lazy)\\s)?(var|let))"
      capture_group: 4
      message: "Please make sure to not use internal access level so that delivery could customize or access this variable.For more information check https://wiki.smithmicro.net/display/SP/iOS+Extensibility+Guidelines"
      severity: error
```
